open_schools_platform/common/selectors.py

```diff
@@ -10,7 +10,9 @@ def selector_factory(model=None):
 
     def selector_wrapper(selector):
         def wrapper(*, filters=None, user: User = None, empty_exception: bool = False,
-                    empty_message: str = f"No such {model_name}", **kwargs):
+                    empty_message: str = f"No such {model_name}", empty_filters: bool = False, **kwargs):
+            if empty_filters and any(arg in filters.values() for arg in ("", None)):
+                return selector(filters=filters).none()
```

Comment:
explain, please

Reply:
In line 14 I check filters. So If empty string or None are present in filters - empty QuerySet is returned.

In line 15 I return empty QuerySet. Unfortunately django does not allow to just return empty QuerySet, so I have to "empty" it with .none() attribute. It is not great but I haven't found a better to return empty queryset(
Maybe you can suggest a solution?

Reply:
give me example of such logic
I remember we want to return empty qs when filters is None, not one filter

Reply:
For example we want to filter queries like this: filters={"recipient_ids": form_ids_string_from_queryset(circles), "sender_ct": ContentType.objects.get(model="studentprofile")}.

Let us assume that "recipient_ids" filter is empty. Then - if we are not returning empty QS, this filter will return all queries with student_profile as sender_ct. And this is not what we want to get.

I actually thought this moment that way - if we are filtering QS with, for example, 2 filters, then it would be strange just to ignore one of them (because it is empty) and return QS filtered by only 1 filter instead of 2.